File: InventoryManagementClient/src/app/models/viewModels/order-view-model.ts
// export class OrderViewModel {
// }
// export interface OrderViewModel {
//     orderID?:number;
//     orderDate?:Date;
//     deliveryDate?:Date;
//     status?:Status;
//     customerID?:number;
//     customerName?:string;
//     orderValue?:number;
// }

import { Status } from "./status";

export class OrderViewModel{
    constructor(
        public orderID?: number,
        public orderDate?: Date,
        public deliveryDate?: Date,
        public status?: Status,
        public customerID?: number,
        public customerName?: string,
        public orderPrice?: number
    ){}
}

File: InventoryManagementAPI/InventoryDatabaseModel/ViewModels/OrderViewModel.cs
using InventoryDatabaseModel.Models;
using Microsoft.AspNetCore.Routing.Constraints;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Security.Principal;
using System.Text;
using System.Threading.Tasks;

namespace InventoryDatabaseModel.ViewModels
{
    public class OrderViewModel
    {
        public int OrderID { get; set; }
        public DateTime OrderDate { get; set; }
        public DateTime? DeliveryDate { get; set; }
        public Status Status { get; set; }
        public int CustomerID { get; set; }
        public string CustomerName { get; set; } = default!;
        public decimal OrderPrice { get; set; }
    }
}
